sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -52,10 +52,10 @@ object TypeCoercion {
       DecimalPrecision ::
       BooleanEquality ::
       StringToIntegralCasts ::
+      Division ::
       FunctionArgumentConversion ::
       CaseWhenCoercion ::
       IfCoercion ::
-      Division ::
```

Comment:
instead of switching orders, can we spend some more time to make these rules non-order-sensitive? thanks